src/Data/List/Base.agda

```diff
@@ -385,6 +385,17 @@ unsnoc as with initLast as
 ------------------------------------------------------------------------
 -- Splitting a list
 
+-- linesBy preserves the empty lines
```

Comment:
I think these comments for `linesBy` and `wordsBy` could be improved. How do "empty lines" and "empty words" relate to the decidable predicate `P`? They probably shouldn't mention whitespace when we're not talking about `String` or `Char`?